File: 50 Coding Challenge Part I/TypeScript/Challenge23.ts
// Challenge 23: Reverse a string

function reverseString(str: string): string {
    let revStr: string = "";
    let len = str.length - 1;

    for (let i = len; i >= 0; i--){
        revStr += str[i];
    }

    return revStr;
}

// Driver Method
console.log(reverseString("JavaScript"));

File: 50 Coding Challenge Part I/dart/lib/challenge23.dart
/*  
  Coding Challenge 23: Reverse an array

  @author: mofirojean
*/

String reverseString(String strs) {
  List<String> reverse = [];
  List<String> letters = strs.trim().split('');

  for (var i = letters.length - 1; i >= 0; i--) {
    reverse.add(letters[i]);
  }

  return reverse.join('');
}
